geomesa-nifi-processors/geomesa-nifi-processors-accumulo/src/test/scala/org/geomesa/nifi/processors/accumulo/PutGeoMesaAccumuloTest.scala

```diff
@@ -0,0 +1,132 @@
+import org.geomesa.nifi.processors.datastore.PutGeoMesaAccumulo
+import org.geotools.data.DataStoreFinder
+import org.junit.{Assert, Test}
+import org.locationtech.geomesa.accumulo.MiniCluster
+import org.locationtech.geomesa.accumulo.data.AccumuloDataStoreParams
+import org.locationtech.geomesa.utils.collection.SelfClosingIterator
+
+class PutGeoMesaAccumuloTest extends LazyLogging {
+
+  import scala.collection.JavaConverters._
+
+  // we use class name to prevent spillage between unit tests
+  lazy val root = s"${MiniCluster.namespace}.${getClass.getSimpleName}"
+
+  // note the table needs to be different to prevent tests from conflicting with each other
+  lazy val dsParams: Map[String, String] = Map(
+    AccumuloDataStoreParams.InstanceIdParam.key -> MiniCluster.cluster.getInstanceName,
+    AccumuloDataStoreParams.ZookeepersParam.key -> MiniCluster.cluster.getZooKeepers,
+    AccumuloDataStoreParams.UserParam.key       -> MiniCluster.Users.root.name,
+    AccumuloDataStoreParams.PasswordParam.key   -> MiniCluster.Users.root.password
+  )
+
+  @Test
+  def testIngest(): Unit = {
+    val catalog = s"${root}Ingest"
+    val runner = TestRunners.newTestRunner(new PutGeoMesaAccumulo())
+    try {
+      dsParams.foreach { case (k, v) => runner.setProperty(k, v) }
+      runner.setProperty(AccumuloDataStoreParams.CatalogParam.key, catalog)
+      runner.setProperty(AbstractGeoIngestProcessor.Properties.IngestModeProp, IngestMode.Converter)
+      runner.setProperty(AbstractGeoIngestProcessor.Properties.SftNameKey, "example")
+      runner.setProperty(AbstractGeoIngestProcessor.Properties.ConverterNameKey, "example-csv")
+      runner.enqueue(getClass.getClassLoader.getResourceAsStream("example.csv"))
+      runner.run()
+      runner.assertTransferCount(AbstractGeoIngestProcessor.Relationships.SuccessRelationship, 1)
+      runner.assertTransferCount(AbstractGeoIngestProcessor.Relationships.FailureRelationship, 0)
+    } finally {
+      runner.shutdown()
+    }
+
+    val ds = DataStoreFinder.getDataStore((dsParams + (AccumuloDataStoreParams.CatalogParam.key -> catalog)).asJava)
+    Assert.assertNotNull(ds)
+    try {
+      val sft = ds.getSchema("example")
+      Assert.assertNotNull(sft)
+      val features = SelfClosingIterator(ds.getFeatureSource("example").getFeatures.features()).toList
+      logger.debug(features.mkString(";"))
+      Assert.assertEquals(3, features.length)
+    } finally {
+      ds.dispose()
+    }
+  }
+
+  @Test
+  def testIngestAttributes(): Unit = {
+    val catalog = s"${root}IngestAttributes"
+    val runner = TestRunners.newTestRunner(new PutGeoMesaAccumulo())
+    try {
+      dsParams.foreach { case (k, v) => runner.setProperty(k, v) }
+      runner.setProperty(AccumuloDataStoreParams.CatalogParam.key, catalog)
+      runner.setProperty(AbstractGeoIngestProcessor.Properties.IngestModeProp, IngestMode.Converter)
+      val attributes = new java.util.HashMap[String, String]()
```

Comment:
Ok, to test the capability to route among SFTs/Converters, would it make sense to add a second SFT+Converter along with a test file? 

In this example, the only SFT and only Converter on the classpath are 'example' and 'example-csv'.  It isn't much of a stretch the only options end up being used;). 

(That said, if a second SFT/converter would be too much work, feel free to push back.)

Optionally, I'd be somewhat amused to see the example.csv data routed to the 'wrong' converter to verify that there is a 'failure'.  

That said, it would be more work, so stop me wherever...

Reply:
eh, we can do that for 3.1 ;)
the processor still requires an explicit config, either through the flowfile attributes or the processor properties. since the properties aren't set here, i think it's effectively testing the case of loading them from the flowfile.